File: php/features.php
<?php
declare(strict_types=1);

// Pony SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class PonyFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new PonyBaseFeature();
            case "test":
                return new PonyTestFeature();
            default:
                return new PonyBaseFeature();
        }
    }
}

File: py/pony_sdk/features.py
# Pony SDK feature factory

from pony_sdk.feature.base_feature import PonyBaseFeature
from pony_sdk.feature.test_feature import PonyTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PonyBaseFeature(),
        "test": lambda: PonyTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
